### PR TITLE
Assemblies and namespaces reorganised

### LINKED ISSUE #517: Revit_Toolkit: Clean up namespace/assembly names
#### Description: 
Assemblies and namespaces sitting in Revit_UI folder to be reshuffled in order to make them more informative. My proposal (discussed to some extent with @al-fisher and @IsakNaslundBh) presented below.

**Project renaming:**
- _Revit_UI_ - to be renamed to _Revit_Listener_ as it is not the entire UI, just listener
- _Engine_Revit_UI_ - to be renamed to _RevitAPI_Engine_ (will allow exposing it in the UIs that work in Revit context)
- _Adapter_Revit_UI_ - to be renamed to _Revit_Adapter_Plugin_ to highlight that it sits on the Revit side (but is not UI)
- projects to be taken out of the _Revit_UI_ folder and placed on the same level as all others

**Namespace renaming:**
- `BH.UI.Revit` - to be renamed to `BH.UI.Revit.Listener`
- `BH.Adapter.Revit` - to stay untouched
- `BH.UI.Revit.Adapter` - to be renamed to `BH.Adapter.Revit.Plugin`
- `BH.oM.Adapters.Revit` - to be renamed to `BH.oM.External.Revit` 
- `BH.Engine.Adapters.Revit` - to be renamed to `BH.Engine.External.Revit`
- `BH.UI.Revit.Engine` - to be renamed to `BH.Engine.External.RevitAPI`
- `BH.oM.Adapters.Revit.Interface` - _Interface_ suffix to be removed, aligned as above

**Class renaming:**
- `RevitUIAdapter` - to be renamed to `RevitAdapterPlugin`

**Folder and file renaming:**
- folder _AdapterActions_ in Revit_UI to be renamed to _EventHandlers_
- _Push.cs_ file in the above folder to be renamed to _PushEvent.cs_
- _Pull.cs_ file in the above folder to be renamed to _PullEvent.cs_

What do you guys think? Any big no from anyone?

### PR DESCRIPTION
**Pull request by pawelbaran**
### Issues addressed by this PR

Closes #517

### Test files

Standard macro tests on [SharePoint](https://burohappold.sharepoint.com/sites/BHoM/02_Current/Forms/AllItems.aspx?RootFolder=%2Fsites%2FBHoM%2F02%5FCurrent%2F12%5FScripts%2F01%5FTest%20Scripts%2FRevit%5FToolkit%2F%5FMacro&FolderCTID=0x0120008122C8891F89054B8ACED0196C70DFC4) - everything should work as before.

**IMPORTANT:** Since this PR changes folder and assembly names, it is recommended to clean the Revit_Toolkit GitHub folder as well as clean the BHoM/Assemblies before testing.

### Changelog

- _Adapter_Revit_UI_ and _Revit_UI_ projects have been merged into a single project named _Revit_Core_Adapter_ with default namespace being `BH.Revit.Adapter.Core`
- _Revit_Engine_UI_ has been renamed to _Revit_Core_Engine_ with default namespace being `BH.Revit.Engine.Core`
- files in the above have been slightly reorganised to make the structure more logical
- references, assemblyInfos, Revit manifest files etc. have been updated following the above
- `SetTags` and `SetIdentifiers` methods were centralized inside _Revit_Core_Engine_, unused messages have been removed
- `RevitUIAdapter` has been renamed to `RevitListenerAdapter`
- `BH.oM.Adapters.Revit.Interface` namespace has been eliminated with all interfaces being moved across to their relevant subnamespaces

### Additional comments
Although the PR is massive, it should not change the mechanics of the code by any means. Thanks to the fact that most of the code base is not exposed by any of the UIs, almost no versioning was needed.

@FraserGreenroyd due to the change in assembly structure, I would expect following issues;
- CI being unhappy about missing assemblies
- Installer being unhappy because of missing assemblies
- End user being unhappy because of the old assemblies not being cleaned up by the installer

Happy to assist with resolving these 🤖

### CODE REVIEW

File: Revit_Core_Adapter/Revit_Core_Adapter.csproj [hunk 1]
```diff
@@ -100,81 +100,41 @@
     <ErrorReport>prompt</ErrorReport>
     <CodeAnalysisRuleSet>ManagedMinimumRules.ruleset</CodeAnalysisRuleSet>
     <TargetFrameworkVersion>v4.7.2</TargetFrameworkVersion>
-    <AssemblyName>Adapter_Revit_UI_2020</AssemblyName>
+    <AssemblyName>Revit_Core_Adapter_2020</AssemblyName>
```
Comment (AndreBalasian) [inline Revit_Core_Adapter/Revit_Core_Adapter.csproj:103]: May be worth adding Revit 2021 at some point

Comment (pawelbaran) [reply Revit_Core_Adapter/Revit_Core_Adapter.csproj:103]: https://github.com/BHoM/Revit_Toolkit/issues/717 there is an issue for that one 👍

**Pull request merged**